boa/src/environment/declarative_environment_record.rs

```diff
@@ -73,32 +75,45 @@ impl EnvironmentRecordTrait for DeclarativeEnvironmentRecord {
                 strict,
             },
         );
-
-        true
+        Ok(())
     }
 
-    fn initialize_binding(&mut self, name: &str, value: Value) {
+    fn initialize_binding(&mut self, name: &str, value: Value) -> Result<(), ErrorKind> {
         if let Some(ref mut record) = self.env_rec.get_mut(name) {
             if record.value.is_none() {
                 record.value = Some(value);
+                Ok(())
             } else {
                 // TODO: change this when error handling comes into play
-                panic!("Identifier {} has already been defined", name);
+                Err(ErrorKind::ReferenceError(format!(
+                    "Identifier {} has already been defined",
+                    name
+                )))
             }
+        } else {
+            panic!(format!("record must have binding for {}", name));
         }
     }
 
     #[allow(clippy::else_if_without_else)]
-    fn set_mutable_binding(&mut self, name: &str, value: Value, mut strict: bool) {
+    fn set_mutable_binding(
+        &mut self,
+        name: &str,
+        value: Value,
+        mut strict: bool,
+    ) -> Result<(), ErrorKind> {
         if self.env_rec.get(name).is_none() {
             if strict {
                 // TODO: change this when error handling comes into play
-                panic!("Reference Error: Cannot set mutable binding for {}", name);
+                return Err(ErrorKind::ReferenceError(format!(
+                    "Reference Error: Cannot set mutable binding for {}",
+                    name
+                )));
```

Comment:
```suggestion
                return Err(ErrorKind::ReferenceError(format!(
                    "Cannot set mutable binding for {}",
                    name
                )));
```
Why is `RefecenceError` mentioned twice